opentelemetry/src/logs/record.rs

```diff
@@ -1,79 +1,46 @@
 use crate::{
-    trace::{SpanContext, SpanId, TraceContextExt, TraceFlags, TraceId},
+    trace::{SpanContext, TraceContextExt},
     Array, Key, StringValue, Value,
 };
 use std::{borrow::Cow, collections::HashMap, time::SystemTime};
 
-#[derive(Debug, Clone)]
-#[non_exhaustive]
-/// LogRecord represents all data carried by a log record, and
-/// is provided to `LogExporter`s as input.
-pub struct LogRecord {
-    /// Event name. Optional as not all the logging API support it.
-    pub event_name: Option<Cow<'static, str>>,
+/// Abstract interface for managing log records. The implementation is provided by the SDK
+/// Abstract interface for log records in an observability framework.
+pub trait LogRecord {
+    /// Sets the creation timestamp.
+    fn set_timestamp(&mut self, timestamp: SystemTime);
 
-    /// Record timestamp
-    pub timestamp: Option<SystemTime>,
+    /// Sets the observed event timestamp.
+    fn set_observed_timestamp(&mut self, timestamp: SystemTime);
 
-    /// Timestamp for when the record was observed by OpenTelemetry
-    pub observed_timestamp: SystemTime,
+    /// Associates a span context.
+    fn set_span_context(&mut self, context: &SpanContext);
 
-    /// Trace context for logs associated with spans
-    pub trace_context: Option<TraceContext>,
+    /// Sets severity as text.
+    fn set_severity_text(&mut self, text: Cow<'static, str>);
 
-    /// The original severity string from the source
-    pub severity_text: Option<Cow<'static, str>>,
-    /// The corresponding severity value, normalized
-    pub severity_number: Option<Severity>,
+    /// Sets severity as a numeric value.
+    fn set_severity_number(&mut self, number: Severity);
 
-    /// Record body
-    pub body: Option<AnyValue>,
+    /// Sets the message body of the log.
+    fn set_body(&mut self, body: AnyValue);
 
-    /// Additional attributes associated with this record
-    pub attributes: Option<Vec<(Key, AnyValue)>>,
-}
-
-impl Default for LogRecord {
-    fn default() -> Self {
-        LogRecord {
-            event_name: None,
-            timestamp: None,
-            observed_timestamp: SystemTime::now(),
-            trace_context: None,
-            severity_text: None,
-            severity_number: None,
-            body: None,
-            attributes: None,
-        }
-    }
-}
+    /// Adds multiple attributes.
+    fn set_attributes(&mut self, attributes: Vec<(Key, AnyValue)>);
```

Comment:
"set" inherently gives the impression that there will be overriding if key is duplicated, making an SDK implementation which does not de-dup bit awkward.. 
Maybe add_attributes, add_attribute are better suited?

Reply:
Agree, and changed accordingly.